guevaraTorres/assignments/twopointers/lc26/lc26.js

```diff
@@ -1,15 +1,75 @@
 // Problem Statement #
-// Given an array of sorted numbers, remove all duplicates from it. You should not use any extra space; after removing the duplicates in-place return the length of the subarray that has no duplicate in it.
+// Given an array of sorted numbers, remove all duplicates from it. You should not use any extra space; 
+// after removing the duplicates in-place return the length of the subarray that has no duplicate in it.
+
+/**
+ * Initialize pointers at the start and end of the array.
+ * If either are duplicates, splice the duplicate from the array
+ * If arr[left] === duplicate, increment left
+ * If arr[right] === duplicate, decrement right   
+ * Return arr.length
+ *  
+ * edge cases => empty array, return -1
+ */
 
 function remove_element(arr, key) {
   // TODO: Write code here
+
+  let left = 0,
+  right = arr.length - 1;
+
+  if(arr.length < 1) return -1;  
+  
+  while(left < right) {
+    if(arr[left] === key) arr.splice(left, 1);
+    if(arr[right] === key) arr.splice(right, 1); right = arr.length - 1;
+    if(arr[left] !== key) left++;  
+  }
+
+  return arr.length;
```

Comment:
Think about the time complexity of this solution. It is O(n^2). Is there a solution that can give us a better time complexity?